.github/workflows/e2e-main.yaml

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: e2e-tests-main
+
+on:
+  push:
+    branches: [main]
+
+  workflow_dispatch:
+    inputs:
+      organization:
+        default: 'redhat-developer'
+        description: 'Organization of the Red Hat Account Extension repository'
+        type: string
+        required: true
+      repositoryName:
+        default: 'podman-desktop-redhat-account-ext'
+        description: 'Podman Desktop RedHat Account Extension repository name'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Red Hat Account Extension repo branch'
+        type: string
+        required: true
+
+jobs:
+  e2e-tests:
+    name: Red Hat Account Extension E2E tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name == 'workflow_dispatch'
+        with:
+          repository: ${{ github.event.inputs.organization }}/${{ github.event.inputs.repositoryName }}
+          ref: ${{ github.event.inputs.branch }}
+          path: ${{ github.event.inputs.repositoryName }}
```

Comment:
So the path maybe different (see line 93)

Reply:
I see your point now. That needs to be fixed.

Reply:
See the step at line 60. I am now using dynamic repository name resolution based on event that triggers the workflow. Variable is used for working-directory now.